### DISCUSSION TITLE
Getting characteristics on all back cameras

### DISCUSSION THREAD

**Discussion by brian7704** [pedroSG94/RootEncoder#1409] [Q&A] [answered]
Is it possible to get the characteristics of each rear camera on devices that have multiple? Also is it possible to tell which one is currently being used or when the user changes between them?

**Comment by pedroSG94** [comment, marked as answer]
Hello,

You can do it by 2 ways:
- Add necessary methods to [Camera2ApiManager](https://github.com/pedroSG94/RootEncoder/blob/master/encoder/src/main/java/com/pedro/encoder/input/video/Camera2ApiManager.java), expose it in RtmpCamera2Base and use RtmpCamera2 (you will need modify the library)

- Use RtmpStream and create your own VideoSource. You can copy this 2 classes:
https://github.com/pedroSG94/RootEncoder/blob/master/encoder/src/main/java/com/pedro/encoder/input/video/Camera2ApiManager.java
https://github.com/pedroSG94/RootEncoder/blob/master/library/src/main/java/com/pedro/library/util/sources/video/Camera2Source.kt
After that, you can use your Camera2SourceCopy that contains Camera2ApiManagerCopy (this last with your methods to get characteristics) in your RtmpStream as VideoSource
